test-packages/test-services/openapi/test-service/api.ts

```diff
@@ -27,5 +27,25 @@ export const TestServiceApi = {
   countEntities: () => new OpenApiRequestBuilder<DefaultApi, 'countEntities'>(
     DefaultApi,
     'countEntities'
+  ),
+  testCaseGetRequiredParameters: (requiredPathItemPathParam: string, simpleTestEntity?: SimpleTestEntity, requiredPathItemQueryParam: string, requiredQueryParam: string, optionalQueryParam?: string, optionalPathItemQueryParam?: string) => new OpenApiRequestBuilder<DefaultApi, 'testCaseGetRequiredParameters'>(
+    DefaultApi,
+    'testCaseGetRequiredParameters',
+    requiredPathItemQueryParam,
+    requiredPathItemPathParam,
+    requiredQueryParam,
+    optionalQueryParam,
+    optionalPathItemQueryParam,
+    simpleTestEntity
+  ),
+  testCasePostRequiredParameters: (requiredPathItemPathParam: string, simpleTestEntity: SimpleTestEntity, requiredPathItemQueryParam: string, requiredQueryParam: string, optionalPathItemQueryParam?: string, optionalQueryParam?: string) => new OpenApiRequestBuilder<DefaultApi, 'testCasePostRequiredParameters'>(
```

Comment:
I am a bit confused by the final singnature. I would have expected from the description an object with keys and some keys parameters are optional. I know that the name is not enough because only name and where it is used (path,query). If I want not to give the `optionalQueryParam` and not `optionalPathItemQueryParam` how would I do that?

Reply:
Something like: 
```
{requiredPath:{paraName1:{},paraName2:{}},requiredQuery:{paraName1:{},paraName2:{}},optionPath:{paraName1?:{},paraName2?:{}}}
```
or do I have a not the right understanding?

Reply:
or are use supposed to fill the unwanted values with `undefinede`?

Reply:
Currently the structure is: `fn(...requiredPathParams, ...requiredBody, ...requiredQueryParams, ...optionalPathParams, ...optionalBody, ...optionalQueryParams)`.

I was planning on doing the following, but I wanted to hear your opinion first:
```
fn(args: {
  requiredParam: string,
  optionalParam?: number
})
```
That way the order is completely irrelevant.

Reply:
Consider duplicates.